cpp/cmake/bbp-setup-pre-commit-config.py

```diff
@@ -8,115 +13,184 @@
 except ImportError:
     from yaml import Loader
 
-from cpplib import str2bool
-
-
 HPC_PRE_COMMITS_REPO_URL = "https://github.com/BlueBrain/hpc-pre-commits"
 DEFAULT_HPC_PRE_COMMIT_REPO = dict(
     repo=HPC_PRE_COMMITS_REPO_URL, rev="master", hooks=[]
 )
-CHECK_CLANG_FORMAT_HOOK_ID = "check-clang-format"
-CHECK_CMAKE_FORMAT_HOOK_ID = "check-cmake-format"
-CHECK_CLANG_TIDY_HOOK_ID = "clang-tidy"
+
+
+class PreCommitConfig:
+    def __init__(self, file):
+        self._file = file
+        if osp.exists(file):
+            with open(file) as istr:
+                self._config = yaml.load(istr, Loader=Loader) or {}
+            self._previous_config = copy.deepcopy(self._config)
+        else:
+            self._config = {}
+            self._previous_config = {}
+        self._bbp_repo = self._initialize_bbp_repo()
+
+    def _initialize_bbp_repo(self):
+        repos = self.config.setdefault("repos", [])
+        for repo in repos:
+            if repo["repo"] == HPC_PRE_COMMITS_REPO_URL:
+                for k, v in DEFAULT_HPC_PRE_COMMIT_REPO.items():
+                    repo.setdefault(k, v)
+                return repo
+        bbp_repo = DEFAULT_HPC_PRE_COMMIT_REPO
+        repos.append(bbp_repo)
+        return bbp_repo
+
+    @property
+    def config(self):
+        return self._config
+
+    def enable_hook(self, new_hook):
```

Comment:
as far as I understand, `enable_hook` is now a function used only internally by `PreCommitConfig`. Thus, in python as convention, these kind of functions are called with a leading "_" so that they cannot be easily called from outside

Reply:
It is indeed an internal function since there are some requirements regarding `new_hook`. Ok to prefix with `_`.